packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -181,6 +181,14 @@ class InspectorPreferencesController extends DisposableController
 
     _customPubRootDirectories.clear();
     await loadCustomPubRootDirectories();
+
+    if (_customPubRootDirectories.value.isEmpty) {
+      // If there are no pub root directories set on the first connection
+      // then try inferring them.
+      final localInspectorService = _inspectorService;
+      await localInspectorService?.inferPubRootDirectoryIfNeeded();
```

Comment:
you can just write `_inspectorService?.inferPubRootDirectoryIfNeeded();` here. Storing in a local doesn't add any value since we are only null checking once either way.